docs/contribute/addnode.qmd

```diff
@@ -0,0 +1,170 @@
+---
+title: "How to add a new node type"
+---
+
+Several parts of the code have to be made aware of the new node type. In the rest of this page we shall call our new node type `NewNodeType`.
+
+# The Julia core
+
+## Parameters
+
+The parameters object (defined in `solve.jl`) passed to the ODE solver must be made aware of the new node type. Therefore define a struct in `solve.jl` which holds the data for each node of the new node type:
+
+```julia
+struct NewNodeType
+    node_id::Vector{Int}
+    # Other fields
+end
+```
+
+This struct should at least contain a vector of all node IDs that are of the new type in a given model. Now you can:
+
+- add `new_node_type::NewNodeType` to the Parameters object;
+- add `new_node_type = NewNodeType(db,config)` to the function `Parameters` in create.jl and add new_node_type add the proper location in the `Parameters` constructor call.
+
+## Reading from configuration
+
+To define a schema for the new node type, add the following to `validation.jl`:
+
+```julia
+@version NewNodeTypeV1 begin
```

Comment:
I see below that NewNodeType has 1 table with the name static. You need 1 schema per table, not per node type. So this would be `NewNodeTypeStaticV1`.

Reply:
In validation.jl there is also `@schema "ribasim.pump.static" PumpStatic`.
And perhaps good to mention that if a specific sorting order is required that this needs to be covered by `sort_by_function` in validation.jl, as well as new_node_type.py. Default sort order is by node ID only.